src/networking.c

```diff
@@ -1583,6 +1583,8 @@ void freeClient(client *c) {
     c->querybuf = NULL;
 
     /* Deallocate structures used to block on blocking ops. */
+    /* If there is any in-flight command, we don't don't record their duration. */
+    c->duration = 0;
```

Comment:
I started running into issues related to module APIs that were calling resetClient() was being called when a client with a pending blocking RM_Call() was getting freed, which triggered the resetClient() call path. There might be a more elegant way to fix it, but I ended up just zero'ing the duration when the client was being freed.

Reply:
i don't understand. are we calling resetClient from within / after freeClient?
same thing for moduleReleaseTempClient?

Reply:
Yes, resetClient is being called from within freeClient, because freeClient calls unbockClient which calls resetClient here: https://github.com/redis/redis/blob/unstable/src/blocked.c#L211.

Reply:
ok.
but now that we fixed the problem with WAIT, can we (at least in theory), replace that assert with a `=0`, and drop many other changes?
i'm ok with to keep the assert, just curious since i'm not looking at this PR with much care..